File: ArraysandStrings/FindClosestNotoZero.java
package ArraysandStrings;

class Main {

    // get the absolute value
    public static float abs(float a){
        return (a <= 0.0F) ? 0.0F-a : a;
    }

    public static int findClosestNumber(int[] nums){
        int closest = nums[0];
        for (int i = 0; i < nums.length; i++){
            if(abs(nums[i]) < abs(closest) || (abs(nums[i]) == abs(closest) && nums[i] > closest)){
                closest = nums[i];
            }
        }
        return closest;
    }


    public static void main(String[] args){

        int[] nums = {-4,-2,1,4,8};
        int closestNumber = findClosestNumber(nums);
        System.out.println(closestNumber);
    }
}

File: ArraysandStrings/FindClosestNotoZero.py
import sys


class Solution():
    def findClosestNumber(self, nums: list[int]) -> int:
        closest = nums[0] 
           # loop through the list and find the smallest positive/absolute value
        for x in nums:
            if abs(x) < abs(closest):
                closest = x
        
        # check if both the positive and negative values are present in the list
        if closest < 0 and abs(closest) in nums:
            return abs(closest)
        else:
            return closest
        
        # time: O(n)
        # space: O(1)
    
    def fasterSolution(self, nums: list[int]) -> int:
        closest= nums[0]

        for x in range(1, len(nums)):
            if abs(nums[x]) < abs(closest) or (abs(nums[x]) == abs(closest) and nums[x] > closest):
                closest = nums[x]
        
        return closest